test/test_nn.py

```diff
@@ -1812,7 +1812,7 @@ def check_weight_norm(l, name, num_params):
 
 
     def test_weight_norm(self):
-        for dtype in [torch.float, torch.bfloat16]:
+        for dtype in [torch.float, torch.bfloat16, torch.float16]:
```

Comment:
This doesn't seem to be covered by optest? We sure it's numerically stable in the reduced precision range?